server/router.js

```diff
@@ -95,5 +96,85 @@ router.get('/', function(req, res) {
   }
 });
 
+let defined = function(variable) {
+  if (typeof variable !== 'undefined' && variable !== null) {
+    return true;
+  }
+  return false;
+};
+
+router.post('/', function(req, res) {
+  var body = '';
+
+  req.on('data', function (data) {
+    body += data;
+    // Too much POST data, kill the connection!
+    // 1e6 === 1 * Math.pow(10, 6) === 1 * 1000000 ~~~ 1MB
+    if (body.length > 1e6) {
+      req.connection.destroy();
+    }
+  });
+
+  req.on('end', function () {
+    var post = qs.parse(body);
+
+    if (defined(post['url']) === false) {
+      res.status(400);
+      res.send('you shall not pass.');
+    } else {
+      let apiUrl = post['url'];
+
+      if (defined(post['referer']) && post['referer']) {
+        if (post['referer'] === "false") {
+          req.headers = u.omit(req.headers, 'referer');
+        } else {
+          req.headers['referer'] = post['referer'];
+        }
+      } else {
+        req.headers = u.omit(req.headers, 'referer');
+      }
+
+      if (defined(post['origin']) && post['origin']) {
+        if (post['origin'] === "false") {
+          req.headers = u.omit(req.headers, 'origin');
+        } else {
+          req.headers['origin'] = post['origin'];
+        }
+      } else {
+        req.headers = u.omit(req.headers, 'origin');
+      }
+
+      if (defined(post['useragent']) && post['useragent']) {
+        if (post['useragent'] === "false") {
+          req.headers = u.omit(req.headers, 'user-agent');
+        } else {
+          req.headers['user-agent'] = post['useragent'];
+        }
+      } else {
+        req.headers = u.omit(req.headers, 'user-agent');
+      }
```

Comment:
I don't really understand why we would want to allow them to pass in custom values...the proxy should just use the headers from the original request.